docs/tools/prism-eslint-hook.js

```diff
@@ -0,0 +1,359 @@
+/**
+ * @fileoverview Use Prism hooks to draw linting errors with red markers on markdown code blocks.
+ * @author Yosuke Ota
+ */
+"use strict";
+
+const Prism = require("prismjs");
+const { docsExampleCodeToParsableCode } = require("./code-block-utils");
+
+let isAvailable = false;
+let Linter = null;
+let astUtils = null;
+
+/*
+ * We can only do the syntax highlighting in the English-language
+ * site because we need the Linter and astUtils. This same
+ * code is run in translation sites where these utilities are
+ * not available, so check to see if they are before attempting
+ * to highlight the code.
+ */
+try {
+    Linter = require("../../lib/api").Linter;
+    astUtils = require("../../lib/shared/ast-utils");
+    isAvailable = true;
+} catch {
+
+    // ignore
+}
+
+/** @typedef {import("../../lib/shared/types").ParserOptions} ParserOptions */
+
+/**
+ * Content that needs to be marked with ESLint
+ * @type {string|undefined}
+ */
+let contentMustBeMarked;
+
+/**
+ * Parser options received from the `::: incorrect` or `::: correct` container.
+ * @type {ParserOptions|undefined}
+ */
+let contentParserOptions;
+
+/**
+ * Set content that needs to be marked.
+ * @param {string} content Source code content that marks ESLint errors.
+ * @param {ParserOptions} options The options used for validation.
+ * @returns {void}
+ */
+function addContentMustBeMarked(content, options) {
+    contentMustBeMarked = content;
+    contentParserOptions = options;
+}
+
+/**
+ * Register a hook for `Prism` to mark errors in ESLint.
+ * @returns {void}
+ */
+function installPrismESLintMarkerHook() {
+
+    /**
+     * A token type for marking the range reported by a rule.
+     * This is also used for the `class` attribute of `<span>`.
+     */
+    const TOKEN_TYPE_ESLINT_MARKED = "eslint-marked";
+
+    /**
+     * Use in the class attribute of `<span>` when an error is displayed in the line-feed, or BOM code.
+     */
+    const CLASS_ESLINT_MARKED_ON_ZERO_WIDTH = "eslint-marked-on-zero-width";
```

Comment:
In the Playground, the corresponding CSS class (`cm-lintPoint`) is also used to mark spans that are literally zero width, e.g. in this incorrect example for `unicode-bom` with option `always`: https://deploy-preview-18041--docs-eslint.netlify.app/rules/unicode-bom#always.

![image](https://github.com/eslint/eslint/assets/6367844/7e7c1ff4-7cb9-4919-8b21-706187e70ab5)

Reply:
Thank you for checking this PR!
I modified it so it looks the same as Playground :+1: